tensornetwork/network.py

```diff
@@ -108,6 +108,40 @@ def add_node(
     self.nodes_set.add(new_node)
     return new_node
 
+  def add_copy_node(
+      self,
+      rank: int,
+      dimension: int,
+      name: Optional[Text] = None,
+      axis_names: Optional[List[Text]] = None,
+      dtype: Type[np.number] = np.float64) -> network_components.CopyNode:
```

Comment:
This reminds me, we should add a `dtype` to TensorNetwork to catch errors early. No changes for now though.